src/main/java/net/minecraftforge/common/capabilities/CapabilityDispatcher.java

```diff
@@ -141,6 +153,47 @@ public void deserializeNBT(CompoundTag nbt)
         }
     }
 
+    @Override
+    public void encode(FriendlyByteBuf out, boolean writeAll)
+    {
+        for (Map.Entry<String, INetworkCapability> entry : networkCapabilities.entrySet())
+        {
+            if (!writeAll && !entry.getValue().requiresSync())
+                continue;
+            out.writeBoolean(true);
+            out.writeUtf(entry.getKey(), 0x100);
```

Comment:
better use a constant for `0x100`, I know it's what vanilla uses but it would increase readability especialy if you ad a javadoc on the constant to explain it.